ogre2/src/Ogre2ThermalCamera.cc

```diff
@@ -362,8 +414,8 @@ void Ogre2ThermalCameraMaterialSwitcher::preRenderTargetUpdate(
 void Ogre2ThermalCameraMaterialSwitcher::postRenderTargetUpdate(
     const Ogre::RenderTargetEvent & /*_evt*/)
 {
-  // restore item to use hlms material
-  for (auto it : this->datablockMap)
+  // restore item to use pbs hlms material
+  for (auto &it : this->datablockMap)
```

Comment:
Is there a reason why we are now taking a reference to the iterator? Based on my understanding, `datablockMap` is a map of pointers, so I don't think there'd be any issues related to using copies vs references.

Reply:
I think I did that out of habit when I'm not sure what the type is. Since it's a pointer, we shouldn't need to do it. I'll revert this change.

Reply:
397a443